src/concrete/ml/sklearn/base.py

```diff
@@ -1328,6 +1339,266 @@ def __init__(self, n_bits: Union[int, Dict[str, int]]):
+            if X is None:
+                if num_unique_thresholds:
+                    min_quantization_value = min_threshold_value
+                    max_quantization_value = max_threshold_value
+                else:
+                    min_quantization_value = 0
+                    max_quantization_value = 1.0
+
+                if num_unique_thresholds == 1:
+                    # If there is only one threshold for this feature
+                    # We want the threshold to be in the middle of a quantization bin
+                    min_quantization_value -= epsilon
+                    max_quantization_value += epsilon
+                elif framework == "xgboost":
+                    # XGBoost uses a < op so we must add a left offset
+                    min_quantization_value -= epsilon
+                else:
+                    # scikit-learn uses =< op so we must add a right offset
+                    max_quantization_value += epsilon
+
+            # Quantizer based on data
+            else:
+                min_quantization_value = X[:, feature_index].min()
+                max_quantization_value = X[:, feature_index].max()
+
+            min_quantization_value = float(min_quantization_value)
+            max_quantization_value = float(max_quantization_value)
+            input_quantizer = QuantizedArray(
+                n_bits=n_bits,
+                values=numpy.array([min_quantization_value, max_quantization_value]),
+            ).quantizer
+            input_quantizers.append(input_quantizer)
+
+        # Convert thresholds to their quantized equivalent
+        quantized_thresholds_array = numpy.empty(bias_1.shape, dtype=numpy.int64)
+        dequantized_thresholds_array = numpy.empty(bias_1.shape, dtype=numpy.float64)
+
+        for threshold_index, threshold_value in enumerate(bias_1[:, 0]):
+            feature_index = int(weight_1[threshold_index, :].argmax())
+            quantized_threshold_value = (
+                input_quantizers[feature_index].quant(threshold_value).astype(numpy.int64)
+            )
+            dequantized_threshold_value = input_quantizers[feature_index].dequant(
+                quantized_threshold_value
+            )
+            quantized_thresholds_array[threshold_index, 0] = quantized_threshold_value
+            dequantized_thresholds_array[threshold_index, 0] = dequantized_threshold_value
+
+        onnx_model.graph.initializer[bias_1_index].CopyFrom(
+            numpy_helper.from_array(
+                quantized_thresholds_array,
+                bias_1_name,
+            )
+        )
+
+        # Tree values pre-processing
+        # i.e., mainly predictions quantization
+        # but also rounding the threshold such that they are now integers
+        model._set_post_processing_params()
+
+        # Get the expected number of ONNX outputs in the sklearn model.
+        expected_number_of_outputs = 1 if is_regressor_or_partial_regressor(model) else 2
```

Comment:
is this always true for classifiers ?

Reply:
Most of this stuff is copy-pasted fromt tree to numpy so I guess it's ok. Also all tests pass so 🤷🏼

Reply:
got it thanks !